### ISSUE TITLE
Broken on GNOME 3.32

### ISSUE THREAD

**Issue by jonafato** [ikalnytskyi/gnome-shell-extension-icon-hider#39]
The extension does not work under 3.32 right now. There are a handful of issues regarding object inheritance and a severe one that occurs during `disable` causing the whole GNOME session to crash (this occurs most noticeably within `_reloadSettings`. Stack trace below:

```
Object St.Bin (0x42a6e90), has been already deallocated — impossible to access it. This might be caused by the object having been destroyed from C code using something such as destroy(), dispose(), or remove() vfuncs.
== Stack trace for context 0x20be2e0 ==
#0        222b688 i   resource:///org/gnome/shell/ui/panel.js:1193 (7f5034f50e50 @ 39)
#1   7ffd4e44fc10 b   resource:///org/gnome/gjs/modules/signals.js:142 (7f50354cb670 @ 386)
#2        222b600 i   resource:///org/gnome/shell/ui/popupMenu.js:872 (7f503511fca0 @ 153)
#3        222b560 i   resource:///org/gnome/shell/ui/popupMenu.js:738 (7f503511f8b0 @ 17)
#4        222b4d8 i   resource:///org/gnome/shell/ui/popupMenu.js:878 (7f503511fdc0 @ 63)
#5        222b450 i   resource:///org/gnome/shell/ui/panelMenu.js:192 (7f5035169c10 @ 53)
#6   7ffd4e451f50 b   self-hosted:977 (7f50351524c0 @ 413)
#7        222b388 i   /home/jonafato/.nix-profile/share/gnome-shell/extensions/icon-hider@kalnitsky.org/extension.js:201 (7f5016e7a820 @ 621)
#8        222b308 i   /home/jonafato/.nix-profile/share/gnome-shell/extensions/icon-hider@kalnitsky.org/extension.js:209 (7f5016e7a8b0 @ 17)
#9   7ffd4e454740 b   self-hosted:979 (7f50351524c0 @ 440)
#10        222b268 i   /home/jonafato/.nix-profile/share/gnome-shell/extensions/icon-hider@kalnitsky.org/extension.js:133 (7f5016e7a670 @ 134)
#11        222b1e0 i   /home/jonafato/.nix-profile/share/gnome-shell/extensions/icon-hider@kalnitsky.org/extension.js:86 (7f5016e7a4c0 @ 59)
#12   7ffd4e4577e0 b   self-hosted:979 (7f50351524c0 @ 440)
#13   7ffd4e4578d0 b   resource:///org/gnome/gjs/modules/signals.js:142 (7f50354cb670 @ 386)
#14        222b150 i   resource:///org/gnome/shell/ui/popupMenu.js:354 (7f503511cee0 @ 56)
#15        222b0c8 i   resource:///org/gnome/shell/ui/popupMenu.js:340 (7f503511ce50 @ 42)
#16        222b040 i   resource:///org/gnome/shell/ui/popupMenu.js:118 (7f503511c1f0 @ 45)
#17   7ffd4e459580 b   self-hosted:979 (7f50351524c0 @ 440)
Object .Gjs_Indicator (0x42a64e0), has been already deallocated — impossible to get any property from it. This might be caused by the object having been destroyed from C code using something such as destroy(), dispose(), or remove() vfuncs.
Object .Gjs_Indicator (0x42a64e0), has been already deallocated — impossible to access it. This might be caused by the object having been destroyed from C code using something such as destroy(), dispose(), or remove() vfuncs.
== Stack trace for context 0x20be2e0 ==
#0   7ffd4e457750 b   resource:///org/gnome/shell/ui/boxpointer.js:268 (7f5035114280 @ 611)
#1   7ffd4e4597a0 b   resource:///org/gnome/shell/ui/layout.js:978 (7f50351010d0 @ 149)
#2   7ffd4e459850 b   self-hosted:975 (7f50351524c0 @ 392)
== Stack trace for context 0x20be2e0 ==
#0   7ffd4e457680 b   resource:///org/gnome/shell/ui/boxpointer.js:482 (7f50351144c0 @ 179)
#1   7ffd4e457750 b   resource:///org/gnome/shell/ui/boxpointer.js:269 (7f5035114280 @ 634)
#2   7ffd4e4597a0 b   resource:///org/gnome/shell/ui/layout.js:978 (7f50351010d0 @ 149)
#3   7ffd4e459850 b   self-hosted:975 (7f50351524c0 @ 392)
segfault at fffffffffffffbb8 ip 00007f504dcd5f38 sp 00007ffd4e456e90 error 5 in libst-1.0.so[7f504dca8000+33000]
Apr 12 23:28:33 nixpad kernel: Code: 00 00 48 8d 3d 67 8c 00 00 e9 e4 2f fd ff 0f 1f 40 00 41 57 41 56 41 55 41 54 55 53 48 83 ec 28 48 63 1d fb 94 01 00 48 01 fb <48> 8b 6b 08 48 85 ed 74 17 48 83 c4 28 48 89 e8 5b 5d 41 5c 41 5d
Unregistered Authentication Agent for unix-session:24 (system bus name :1.1253, object path /org/freedesktop/PolicyKit1/AuthenticationAgent, locale en_US.UTF-8) (disconnected from bus)
WARNING: Application 'org.gnome.Shell.desktop' killed by signal 11
WARNING: App 'org.gnome.Shell.desktop' respawning too quickly
WARNING: Application 'org.gnome.Shell.desktop' killed by signal 11
Unrecoverable failure in required component org.gnome.Shell.desktop
```

**Comment by theychx** [comment]
> and a severe one that occurs during disable causing the whole GNOME session to crash

I found this commit in the repo of the standard suite of extensions:
https://gitlab.gnome.org/GNOME/gnome-shell-extensions/commit/ee85839d607ca71a9e3b8e7a356d83f79d407d96

**Comment by jonafato** [comment]
@theychx Thank you! I'll get the same fix added to my PR. I grabbed a bunch of things from that repo, but as I'm sure you know, that's a frustrating bug to try to fix without knowing the underlying cause.

**Comment by jonafato** [comment]
Fixed by #40.
